server/game/core/ability/abilityTargets/CardTargetResolver.ts

```diff
@@ -17,8 +17,16 @@ export class CardTargetResolver extends TargetResolver<ICardTargetResolver<Abili
     private selector;
     public constructor(name: string, properties: ICardTargetResolver<AbilityContext>, ability: PlayerOrCardAbility) {
         super(name, properties, ability);
+
         this.selector = this.getSelector(properties);
 
+        // a player can always choose not to pick a card from a zone that is hidden from their opponents (SWU Comp Rules 2.0 1.17.4)
+        if (this.allZonesAreHidden(this.properties.locationFilter)) {
+            this.properties.optional = true;
```

Comment:
So I did a quick debug and it looks like code execution never makes it inside this block? I think it's because `locationFilter` by default is `WildcardLocation.AnyAttackable` which isn't covered by `EnumHelpers.isHidden`. That's something I should've caught, it needs to handle wildcards correctly as well.

Can you fix and then add a test case that confirms that the prompt title has the appended text from this code block

Reply:
It does go into the block in the tests for ECL and Chewie; not Ozzel though, I think because he uses a SelectCardSystem instead of a target resolver. I'll add a test though.